src/app/commands/cmd_layer_visibility.cpp

```diff
@@ -45,16 +45,31 @@ bool LayerVisibilityCommand::onEnabled(Context* context)
 bool LayerVisibilityCommand::onChecked(Context* context)
 {
   const ContextReader reader(context);
-  const Layer* layer = reader.layer();
-  return (layer && layer->isVisible());
+  const SelectedLayers& selLayers = reader.site()->selectedLayers();
+  bool visible = false;
+  for (auto layer : selLayers)
+  {
+    if(layer && layer->isVisible())
+      visible = true;
+  }
+  return visible;
 }
 
 void LayerVisibilityCommand::onExecute(Context* context)
 {
   ContextWriter writer(context);
-  Layer* layer = writer.layer();
+  const SelectedLayers& selLayers = writer.site()->selectedLayers();
```

Comment:
Don't worry, you can now copy the lines, I'll see how to do it right (or if you want, you can then create then another PR for that refactor).